### ISSUE TITLE
Sequelize (incorrectly) assumes database date is UTC

### ISSUE THREAD

**Issue by hdpinto** [sequelize/sequelize#7243]
Trying to query a date from the database results in the date (localized to the current time zone in the database) being read as UTC (-00:00). For example, querying the following date in EST:

| row_id | action_date                     | ... |
|--------|--------------------------|---|
| 1      |  2017-02-13T17:35:57.000 | ... |

Would come back as `2017-02-13T17:35:57.000Z`, the Z being added without taking into account the fact that there are 5 hours between UTC and EST.

I've searched around and initialized the database with the timezone option, as per the recommendation in [#2590](https://github.com/sequelize/sequelize/issues/2590), but this seems to have no effect on the output.

```javascript
var Sequelize = require('sequelize');

var options = {
  host: '127.0.0.1',
  port: '1433',
  dialect: 'mssql',
  //...
  timezone: '-05:00'
}

var sql = new Sequelize('databaseName', 'username', process.env.PASSWORD, options)
```

The query was a simple `Model.findAll({ ... })`, and I was expecting the dates to either be appropriately localized (`2017-02-13T17:35:57.000-05:00`) , or be properly converted to UTC (`2017-02-13T22:35:57.000Z`).

__Dialect:__ mssql
__tedious version:__ 1.14.0
__MSSQL Server:__ 12.0.2000
__Sequelize version:__ 3.27.0

**Comment by hdpinto** [comment]
Since this unwanted behavior happens pretty reliably (no matter what date I use), I'm using the following really dirty workaround on the front end:

```javascript
function checkDate(item){
  let tmp = item.date.toString();

  if (tmp.includes("Z"))
    tmp = tmp.substring(0,  tmp.indexOf("Z")); // Removes the trailing Z off of the date

  let now = moment();
  let dateToCheck = moment(tmp);

  // rest of function
}
```

I _really_ don't like having to do this, because it breaks time zones and properly formatted dates from other sources, and would rather have the dates come in properly formatted in the first place.

**Comment by harshithkashyap** [comment]
@hdpinto Sequelize used [`DATETIME2`](https://github.com/sequelize/sequelize/blob/v3/lib/dialects/mssql/data-types.js#L96-L107) MSSQL data type which does not take into consideration of timezone. This has been fixed on v4/master branch by using [`DATETIMEOFFSET`](https://github.com/sequelize/sequelize/pull/7201) datatype. Suggest you to upgrade to master branch or wait for the stable release.

**Comment by NivethikaM** [comment]
I am getting this issue in V5. I have used Date datatype and it is converted into DateTimeOffset (MSSQL) in SQL SERVER.

I have provided timezone as '+05:30' and it is storing properly. But when I am retrieving from Database the value is UTC date.

In dialect option I have set useUTC: false, DateString, typecast but it is of no use.

Could anyone help me on how to retrieve the date value as it is in database?
